AudioOutput.h

```diff
@@ -46,6 +46,7 @@
 
 #ifndef AUDIOOUTPUT_H
 #define AUDIOOUTPUT_H
+#include <FixMath.h>
```

Comment:
On a somewhat general note, I wonder, whether we can still avoid a hard dependency on the - now external - FixMath. Sure, installing it via the library manager should be a breeze, but at least when installing from git, that may be a small bonus.

The trick would be to never have a *full* specialization. Due to the RANGE parameter, I think that requirement is actually easy to meet. Pseudocode:

```
//#include <FixMath.h> -> removed
template<int8_z, int8_t, uint64_t> class SFix;  // forward declaration, only

// This partial specialization is only compiled when actually used
// therefore it's ok that SFix is an incomplete type, so far.
template<int8_t NF, uint64_t RANGE> void someFunction(SFix<16,NF,RANGE> x) { [...] };
```

Reply:
And to clarify, FixMath.h would then have to be included, manually, in sketches that actually use it.

Reply:
That's a good idea, I did not think of it. My only concern would be that the dependency is less clear (ie, the error message would be something like: `SFix is declared but never defined` and not `File FixMath.h not found`). But at the same time, the error message for the examples would be clear. Will try to see how that goes!

Reply:
I might be overlooking something, but I am not sure this is always simple. Let's take this line from Oscil.h:

```
        template <int8_t NI, int8_t NF, uint64_t RANGE>
        inline
        void setFreq(UFix<NI,NF,RANGE> frequency)
        {
        setFreq_Q16n16(UFix<16,16>(frequency).asRaw());
        }
```

with the forward declaration `setFreq(UFix<NI,NF,RANGE> frequency)` is correctly resolved by the compiler but `UFix<16,16>(frequency)` is not as it does not have the correct number of arguments. Adding another argument, say `_RANGE`, is not straightforward because we would need to add in Mozzi the macros to compute the default range here (hence negating a bit the advantage of having it with a default value in FixMath.

Did I overlooked something?

Reply:
> Did I overlooked something?

Only the fact that I tend to comment before testing... It may or may not be possible to weasle our way out of this particular problem using something like

```
setFreq_Q16n16((int32_t) frequency.sL(16-NF).asRaw());
```

But I'm not sure, where the next problem of this type pops up, and I'll admit, I'm not totally convinced, myself.